src/main/scala/treadle/executable/SymbolTable.scala

```diff
@@ -210,12 +210,7 @@ object SymbolTable extends LazyLogging {
     var printfCardinal: Int = 0
     val lastPrintfInMOdule = new mutable.HashMap[Module, Symbol]
 
-    val moduleMemoryToMemorySymbol = new mutable.HashMap[String, mutable.HashSet[Symbol]] {
-      override def default(key: String): mutable.HashSet[Symbol] = {
-        this(key) = new mutable.HashSet[Symbol]()
-        this(key)
-      }
-    }
+    val moduleMemoryToMemorySymbol = new mutable.HashMap[String, mutable.HashSet[Symbol]]()
```

Comment:
The deprecation warning (an error when building the docs) says that `withDefaultValue` is the normal way to fix this, but it does **not** mesh with mutation. `withDefaultValue` never actually adds the key to the underlying `mutable.HashMap` so it's just sort of a mess, don't use it with mutable stuff.

Reply:
I think I've used this idiom in quite a few places. Too bad `withDefaultValue` doesn't work in the obvious way

Reply:
Yeah it is a powerful tool to lose. One option is to have a custom class extend `trait mutable.Map` that is just backed up by a `mutable.HashMap`. There are 4 methods that we'd have to implement, but it's an option for cases where just switching to `getOrElseUpdate` is too onerous. In this case it was only a few places so I thought it was fine.